packages/foundry/contracts/hooks/VolatilityFee/VolatilityFeeHookV2.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import { IHooks } from "@balancer-labs/v3-interfaces/contracts/vault/IHooks.sol";
+import { IRouter } from "@balancer-labs/v3-interfaces/contracts/vault/IRouter.sol";
+import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
+import {
+    AfterSwapParams,
+    LiquidityManagement,
+    SwapKind,
+    TokenConfig,
+    HookFlags,
+    PoolSwapParams
+} from "@balancer-labs/v3-interfaces/contracts/vault/VaultTypes.sol";
+
+import { EnumerableMap } from "@balancer-labs/v3-solidity-utils/contracts/openzeppelin/EnumerableMap.sol";
+import { FixedPoint } from "@balancer-labs/v3-solidity-utils/contracts/math/FixedPoint.sol";
+import { VaultGuard } from "@balancer-labs/v3-vault/contracts/VaultGuard.sol";
+
+import { BaseHooks } from "@balancer-labs/v3-vault/contracts/BaseHooks.sol";
+import { IBasePoolFactory } from "@balancer-labs/v3-interfaces/contracts/vault/IBasePoolFactory.sol";
+
+contract VolatilityFeeHookV2 is BaseHooks, Ownable, VaultGuard {
+    using FixedPoint for uint256;
+    address private immutable _trustedRouter;
+    address private immutable _allowedFactory;
+
+
+    // Constructor
+    // Add Trusted Roter and Allowed Factory for the Hook at Deployment
+    constructor(IVault vault,address allowedFactory, address router) VaultGuard(vault) Ownable(msg.sender) {
+        _trustedRouter = router;
```

Comment:
It looks like `_trustedRouter` is unused, any reason why it was included here?

Reply:
I took the boilerplate code from `ExampleHooks` and must have missed it since my linter wasn’t on. It’s not particularly useful in its current state, but it could add an extra layer of security if implemented in `onComputeDynamicSwapFeePercentage`, similar to how it’s used in `veBALFeeDiscountHook.sol`.